src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -839,7 +839,7 @@ class Peer(
     }
 
     private suspend fun processIncomingPayment(item: Either<PayToOpenRequest, UpdateAddHtlc>) {
-        val currentBlockHeight = currentTipFlow.filterNotNull().first().first
+        val currentBlockHeight = currentTipFlow.filterNotNull().first()
```

Comment:
This one is risky because it is needed *after* having accepted and cross-signed an htlc. In a scenario where we cannot access the blockchain this could lead to a payment timeout. We should probably reject immediately if we don't have a blockheight, instead of hanging.

Reply:
Good point, and this should be quite easy to do, with something like:

```kotlin
when (val currentBlockHeight = currentTipFlow.value) {
    null -> // fail HTLC immediately
    else -> {
        // current processing
    }
}
```

But this could also negatively impact payment success: background payments may fail because we're simply slow to receive the block height, whereas if we waited a bit more on the `currentTipFlow` the background payment could have succeeded. And that's true for non-background payments as well when getting the block height is slow but eventually works.

If we decide to do this, it should probably be a separate follow-up PR with a dedicated unit test.

Reply:
We could use a 30s timeout here to handle the "laggy" scenario. But it's starting to look a bit ugly.

By the way, simply failing the htlc is already kind of ugly due to the htlc/pay-to-open cases. It's feasible but not as straightforward as it seems.